Content.Server/Fluids/EntitySystems/SpraySystem.cs

```diff
@@ -106,21 +103,21 @@ private void OnAfterInteract(EntityUid uid, SprayComponent component, AfterInter
             var vapor = Spawn(component.SprayedPrototype, vaporPos);
             var vaporXform = xformQuery.GetComponent(vapor);
 
-            vaporXform.WorldRotation = rotation;
+            _transform.SetWorldRotation(vaporXform, rotation);
 
             if (TryComp(vapor, out AppearanceComponent? appearance))
             {
-                _appearance.SetData(uid, VaporVisuals.Color, solution.GetColor(_proto).WithAlpha(1f), appearance);
-                _appearance.SetData(uid, VaporVisuals.State, true, appearance);
+                _appearance.SetData(vapor, VaporVisuals.Color, solution.GetColor(_proto).WithAlpha(1f), appearance);
+                _appearance.SetData(vapor, VaporVisuals.State, true, appearance);
             }
```

Comment:
This if-statement is kind of redundant, but maybe I should just log when it can't find an `AppearanceComponent`? Not sure why `SetData()` doesn't just allow me to pass an optional `logMissing`